pilot/pkg/model/context.go

```diff
@@ -513,6 +513,9 @@ type NodeMetadata struct {
 	// AutoRegister will enable auto registration of the connected endpoint to the service registry using the given WorkloadGroup name
 	AutoRegisterGroup string `json:"AUTO_REGISTER_GROUP,omitempty"`
 
+	// WorkloadEntryName is the name of the workloadentry the proxy is associated with
+	WorkloadEntryName string `json:"WORKLOAD_ENTRY_NAME,omitempty"`
```

Comment:
I think the serviceentry registry creates an index of `ip/namespace->wle`. Seems like it could be a much nicer experience if the user didn't have to specify this. Not saying theres anything wrong with leaving this in, but we _should_ be able to automate it a bit more.

Reply:
https://github.com/istio/istio/blob/a1124ddbebac51af0394b1f4ccfd71db5b8492b2/pilot/pkg/serviceregistry/kube/controller/controller.go#L244

It's actually in the kube registry.. but we should be able to use something like this.. probably also want namespace.

Reply:
we can search for <ip,network> pair.

I would agree we probably want to make this automatic

Reply:
Search in config store? I feel like it should be indexed with an informer at the very least, or it will be a linear search each time.

Reply:
I agree, I meant we should index it by ip,network and then lookup by the same